src/config.rs

```diff
@@ -1889,15 +1889,15 @@ impl TiKvConfig {
         self.storage.validate()?;
 
         self.raft_store.region_split_check_diff = self.coprocessor.region_split_size / 16;
+
+        let store_path = Path::new(&self.storage.data_dir).to_owned();
         self.raft_store.raftdb_path = if self.raft_store.raftdb_path.is_empty() {
-            config::canonicalize_sub_path(&self.storage.data_dir, "raft")?
+            format!("{}", store_path.join("raft").display())
```

Comment:
The reason to `canonicalize_sub_path` is that "data" and "data/../data" won't become two different path, otherwise it will make TiKV fail to be restarted.